File: backend/test/domain/project_mother.py
import uuid

from app.domain.project.project import Project


class ProjectMother:

    @classmethod
    def create(cls,
               project_name="project_name",
               project_uri="http://localhost:8080",
               authentication=True,
               username="username",
               password="password"):
        return Project(
            project_id=str(uuid.uuid4()),
            project_name=project_name,
            project_uri=project_uri,
            authentication=authentication,
            username=username,
            password=password
        )

File: frontend/test/app/core/model/ProjectMother.ts
import {Project} from "../../../../src/app/core/model/Project";


export class ProjectMother {

  public static create(data = {
    project_id: '123',
    project_name: 'project name',
    project_uri: 'http://localhost',
    authentication: true,
    username: 'root',
    password: 'root'
  }): Project {
    return {
      project_id: data.project_id,
      project_name: data.project_name,
      project_uri: data.project_uri,
      authentication: data.authentication,
      username: data.username,
      password: data.password
    }
  }
}
